backend/business-partner-agent/src/main/java/org/hyperledger/bpa/impl/InvitationParser.java

```diff
@@ -0,0 +1,193 @@
+        HttpUrl url = HttpUrl.parse(URLDecoder.decode(invitationUrl, StandardCharsets.UTF_8));
+        String invitationBlock = null;
+        if (url != null) {
+            invitationBlock = parseInvitationBlock(url);
+            if (StringUtils.isEmpty(invitationBlock)) {
+                invitationBlock = parseInvitationBlockFromRedirect(url);
+            }
+            if (StringUtils.isEmpty(invitationBlock)) {
+                throw new InvitationException("Invitation Url does not contain a known or valid invitation.");
+            }
+
+            Invitation invite = parseInvitation(invitationBlock);
+
+            if (StringUtils.isNotEmpty(invite.getError())) {
+                throw new InvitationException(invite.getError());
+            } else {
+                // right now there is only one option for a success
+                if (invite.isParsed() && invite.getInvitationRequest() != null) {
+                    ReceiveInvitationRequest r = invite.getInvitationRequest();
+                    return CheckInvitationResponse.builder()
+                            .label(r.getLabel())
+                            .invitation(invite.getInvitation())
+                            .invitationBlock(invite.getInvitationBlock())
+                            .build();
+                }
+            }
+        } else {
+            throw new InvitationException("Invitation Url could not be decoded. Cannot determine invitation details.");
+        }
+        return null;
+    }
+
+    public Invitation parseInvitation(String invitationBlock) {
+        Invitation invitation = new Invitation();
+        if (StringUtils.isNotEmpty(invitationBlock)) {
+
+            invitation.setInvitationBlock(invitationBlock);
+
+            String decodedBlock = decodeInvitationBlock(invitationBlock);
+            if (StringUtils.isNotEmpty(decodedBlock)) {
+                Map<String, Object> o = null;
+                try {
+                    o = mapper.readValue(decodedBlock, HashMap.class);
+                    invitation.setInvitation(o);
+                    invitation.setParsed(true);
+
+                    if ("did:sov:BzCbsNYhMrjHiqZDTUASHg;spec/connections/1.0/invitation".equals(o.get("@type"))) {
+                        // Invitation
+                        Gson gson = new Gson();
+                        ReceiveInvitationRequest r = gson.fromJson(decodedBlock, ReceiveInvitationRequest.class);
+                        invitation.setInvitationRequest(r);
+
+                    } else if ("did:sov:BzCbsNYhMrjHiqZDTUASHg;spec/out-of-band/1.0/invitation"
+                            .equals(o.get("@type"))) {
+                        invitation.setOob(true);
+                        // not supporting this until we can parse and send to aries client successfully
+                        // coming very soon with next update of Aries Client Jar!!!
+                        String msg = "Out of band Invitations are currently not supported";
```

Comment:
I have just released a new client 0.7.4 with the fix

Reply:
ok, I'll pull that in and add this functionality. Thanks for getting that done so quickly.